sdk/evaluation-context.json

```diff
@@ -0,0 +1,147 @@
+{
+    "$defs": {
+        "EnvironmentEvaluationContext": {
+            "properties": {
+                "api_key": {
+                    "title": "Api Key",
+                    "type": "string"
+                }
+            },
+            "required": [
+                "api_key"
+            ],
+            "title": "EnvironmentEvaluationContext",
+            "type": "object"
+        },
+        "FeatureEvaluationContext": {
+            "properties": {
+                "name": {
+                    "title": "Name",
+                    "type": "string"
+                }
+            },
+            "required": [
+                "name"
+            ],
+            "title": "FeatureEvaluationContext",
+            "type": "object"
+        },
+        "IdentityEvaluationContext": {
+            "properties": {
+                "identifier": {
+                    "title": "Identifier",
+                    "type": "string"
+                },
+                "traits": {
+                    "additionalProperties": {
+                        "anyOf": [
+                            {
+                                "$ref": "#/$defs/TraitEvaluationContext"
+                            },
+                            {
+                                "type": "null"
+                            }
+                        ]
+                    },
+                    "title": "Traits",
+                    "type": "object"
+                },
+                "transient": {
+                    "anyOf": [
+                        {
+                            "type": "boolean"
+                        },
+                        {
+                            "type": "null"
+                        }
+                    ],
+                    "default": null,
+                    "title": "Transient"
+                }
+            },
+            "required": [
+                "identifier",
+                "traits"
+            ],
+            "title": "IdentityEvaluationContext",
+            "type": "object"
+        },
+        "TraitEvaluationContext": {
+            "properties": {
+                "value": {
+                    "title": "Value",
+                    "type": "string"
+                },
+                "type": {
+                    "$ref": "#/$defs/TraitValueType"
+                },
+                "transient": {
+                    "anyOf": [
+                        {
+                            "type": "boolean"
+                        },
+                        {
+                            "type": "null"
```

Comment:
Why do we need null?

Reply:
To distinguish between a default value and field presence. We have a little more flexibility this way. This approach is taken for all "optional" fields here.

Reply:
For example, we might consider adding a per-org/per-project option to regard all incoming traits as transient by default. If we offered a default `false` value, this would render the schema semantics inaccurate.